lib/Sema/TypeCheckAttr.cpp

```diff
@@ -653,6 +654,7 @@ class AttributeChecker : public AttributeVisitor<AttributeChecker> {
     IGNORED_ATTR(Testable)
     IGNORED_ATTR(WarnUnqualifiedAccess)
     IGNORED_ATTR(ShowInInterface)
+    IGNORED_ATTR(DiscardableResult)
```

Comment:
Should the `@discardableResult` attribute be allowed on Void functions?

Currently `@warn_unused_result` is allowed for that case, but I'm not sure it makes sense for the new attribute.